src/studio-home/factories/mockApiResponses.jsx

```diff
@@ -48,36 +48,32 @@ export const generateGetStudioHomeDataApiResponse = () => ({
   allowToCreateNewOrg: false,
 });
 
+/** Mock for the deprecated /api/contentstore/v1/home/courses endpoint. Note this endpoint is NOT paginated. */
 export const generateGetStudioCoursesApiResponse = () => ({
-  count: 5,
-  next: null,
-  previous: null,
-  numPages: 2,
-  results: {
-    courses: [
-      {
-        courseKey: 'course-v1:HarvardX+123+2023',
-        displayName: 'Managing Risk in the Information Age',
-        lmsLink: '//localhost:18000/courses/course-v1:HarvardX+123+2023/jump_to/block-v1:HarvardX+123+2023+type@course+block@course',
-        number: '123',
-        org: 'HarvardX',
-        rerunLink: '/course_rerun/course-v1:HarvardX+123+2023',
-        run: '2023',
-        url: '/course/course-v1:HarvardX+123+2023',
-      },
-      {
-        courseKey: 'org.0/course_0/Run_0',
-        displayName: 'Run 0',
-        lmsLink: null,
-        number: 'course_0',
-        org: 'org.0',
-        rerunLink: null,
-        run: 'Run_0',
-        url: null,
-      },
-    ],
-    inProcessCourseActions: [],
-  },
+  archivedCourses: /** @type {any[]} */([]),
+  courses: [
+    {
+      courseKey: 'course-v1:HarvardX+123+2023',
+      displayName: 'Managing Risk in the Information Age',
+      lmsLink: '//localhost:18000/courses/course-v1:HarvardX+123+2023/jump_to/block-v1:HarvardX+123+2023+type@course+block@course',
+      number: '123',
+      org: 'HarvardX',
+      rerunLink: '/course_rerun/course-v1:HarvardX+123+2023',
+      run: '2023',
+      url: '/course/course-v1:HarvardX+123+2023',
+    },
+    {
+      courseKey: 'org.0/course_0/Run_0',
+      displayName: 'Run 0',
+      lmsLink: null,
+      number: 'course_0',
+      org: 'org.0',
+      rerunLink: null,
+      run: 'Run_0',
+      url: null,
+    },
+  ],
+  inProcessCourseActions: [],
```

Comment:
This mock was totally wrong. This endpoint is never paginated. I confirmed in my browser, and [you can see in the code for the endpoint](https://github.com/openedx/edx-platform/blob/ca11c14c58f7f522a265f6e777b283a0f35e030e/cms/djangoapps/contentstore/rest_api/v1/views/home.py#L93-L165), and that's also what [it says in this MFE's code](https://github.com/openedx/frontend-app-authoring/blob/9d3a05f1bd27be694108a15274dddcca1d7950ae/src/studio-home/data/thunks.js#L35-L40).

Before my refactor here, the tests were working around this issue by mocking the endpoint with just the `results` field from this data, rather than using the whole data object. But it's confusing that the mock contains pagination when the real endpoint doesn't, so I've fixed that here.